src/test/java/com/facebook/openwifirrm/modules/ModelerUtilsTest.java

```diff
@@ -547,4 +551,290 @@ void testPostDot11nAggregatedWifiScanEntry() {
+				.get(0).rssi,
+			new ArrayList<>(Arrays.asList(10, 20, 30))
+		);
+
+		assertEquals(
+			bssidToAggregatedStates
+				.get(TestUtils.getBssidStationKeyPair(bssidA, stationA1))
+				.get(1).rssi,
+			new ArrayList<>(Arrays.asList(40, 50))
+		);
+		assertEquals(
+			bssidToAggregatedStates
+				.get(TestUtils.getBssidStationKeyPair(bssidA, stationA2))
+				.get(0).rssi,
+			new ArrayList<>(Arrays.asList(20, 30, 40, 60))
+		);
+
+		State toBeAggregated2 = TestUtils.createState(
+			3,
+			20,
+			200,
+			bssidB,
+			new String[] { stationB },
+			new int[] { 40 },
+			1
+		);
+		ModelerUtils
+			.addStateToAggregation(bssidToAggregatedStates, toBeAggregated2);
+		assertEquals(
+			bssidToAggregatedStates
+				.get(TestUtils.getBssidStationKeyPair(bssidB, stationB))
+				.get(0).rssi,
+			new ArrayList<>(Arrays.asList(10, 20, 30, 40))
+		);
+
+	}
+
+	@Test
+	void testGetAggregatedStates() {
+		final long obsoletionPeriodMs = 60000000;
+		final String serialNumberA = "aaaaaaaaaaaa";
+		final String bssidA = "aa:aa:aa:aa:aa:a";
+		final String serialNumberB = "bbbbbbbbbbbb";
+		final String bssidB = "bb:bb:bb:bb:bb:bb";
+		final String serialNumberC = "cccccccccccc";
+		final String bssidC = "cc:cc:cc:cc:cc:cc";
+
+		long refTimeMs = TestUtils.DEFAULT_LOCAL_TIME;
+
+		DataModel dataModel = new DataModel();
+
+		// This serie of StateA is used to test a valid input states.
+		State time1ToStateA = TestUtils.createState(
+			1,
+			80,
+			1000,
+			bssidA,
+			new String[] { "stationA1", "stationA2", "stationA2", "stationA3" },
+			new int[] { -84, -67, -67, 10 },
+			2,
+			40,
+			2000,
+			bssidA,
+			new String[] { "stationA1" },
+			new int[] { -80 },
+			TestUtils.DEFAULT_LOCAL_TIME
+		);
+
+		State time2ToStateA = TestUtils.createState(
+			1,
+			80,
+			1000,
+			bssidA,
+			new String[] { "stationA1", "stationA3" },
+			new int[] { 27, 100 },
+			2,
+			40,
+			2000,
+			bssidA,
+			new String[] { "stationA2", "stationA2" },
+			new int[] { 180, 67 },
+			TestUtils.DEFAULT_LOCAL_TIME - 800
+		);
+
+		//As State time3ToStateA is obsolete, it should not be aggregated.
+		State time3ToStateA = TestUtils.createState(
+			1,
+			80,
+			1000,
+			bssidA,
+			new String[] { "stationA1", "stationA2", "stationA4" },
+			new int[] { 24, 27, 1000 },
+			2,
+			40,
+			2000,
+			bssidA,
+			new String[] { "stationA1", "staionA2" },
+			new int[] { 180, 180 },
+			// Set the localtime exactly obsolete
+			TestUtils.DEFAULT_LOCAL_TIME - obsoletionPeriodMs - 1
+		);
+
+		dataModel.latestStates.put(
+			serialNumberA,
+			new ArrayList<>(
+				Arrays.asList(
+					time1ToStateA,
+					time2ToStateA,
+					time3ToStateA
+				)
+			)
+		);
+
+		Map<String, Map<String, List<AggregatedState>>> aggregatedMap =
+			ModelerUtils
+				.getAggregatedStates(dataModel, obsoletionPeriodMs, refTimeMs);
+
+		assertEquals(
+			aggregatedMap.get(serialNumberA).get(TestUtils.getBssidStationKeyPair(bssidA, "stationA1")).size(),
+			2
+		);
```

Comment:
Did you run the formatter? This line is obviously over 80 chars.

Reply:
Yes I have run the formatter but it doesn't change anything. Do I need to format it manually?

Reply:
Seems that `spotless` does not enforce the line width? @elludraon  If so, I think there's no need to do it manually. but I wonder if your IDE or editor could do it for you.